From 6ef777e9a022bee2a80fa671118e7e2657e52693 Mon Sep 17 00:00:00 2001
From: Jonathan Desrosiers <desrosj@git.wordpress.org>
Date: Wed, 10 Jun 2020 17:44:37 +0000
Subject: [PATCH] Formatting: Ensure that wp_validate_redirect() sanitizes a
 wider variety of characters.

Merges [47949] to the 5.4 branch.
Props: xknow, vortfu.

git-svn-id: https://develop.svn.wordpress.org/branches/5.4@47955 602fd350-edb4-49c9-b593-d223f7449a82
---
 src/wp-includes/pluggable.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wp-includes/pluggable.php b/src/wp-includes/pluggable.php
index 3de1934044d..ffc061dccea 100644
--- a/src/wp-includes/pluggable.php
+++ b/src/wp-includes/pluggable.php
@@ -1407,7 +1407,7 @@ function wp_safe_redirect( $location, $status = 302, $x_redirect_by = 'WordPress
 	 * @return string redirect-sanitized URL
 	 */
 	function wp_validate_redirect( $location, $default = '' ) {
-		$location = trim( $location, " \t\n\r\0\x08\x0B" );
+		$location = wp_sanitize_redirect( trim( $location, " \t\n\r\0\x08\x0B" ) );
 		// Browsers will assume 'http' is your protocol, and will obey a redirect to a URL starting with '//'.
 		if ( substr( $location, 0, 2 ) == '//' ) {
 			$location = 'http:' . $location;
